libs/gi/page-team/src/CharacterDisplay/CustomMultiTarget/MTargetEditor.tsx

```diff
@@ -0,0 +1,353 @@
+import {
+  Box,
+  CardActionArea,
+  Collapse,
+  Divider,
+  Grid,
+  IconButton,
+  InputAdornment,
+  MenuItem,
+  Typography,
+  useMediaQuery,
+  useTheme,
+} from '@mui/material'
+import { useCallback, useContext, useMemo, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import OptimizationTargetSelector from '../Tabs/TabOptimize/Components/OptimizationTargetSelector'
+
+const keys = [...allInputPremodKeys]
+const wrapperFunc = (e: JSX.Element, key?: string) => (
+  <Grid item key={key} xs={1}>
+    {e}
+  </Grid>
+)
+
+export default function MTargetEditor({
+  customTarget,
+  setCustomTarget,
+  deleteCustomTarget,
+  rank,
+  maxRank,
+  setTargetIndex,
+  onDup,
+}: {
+  customTarget: CustomTarget
+  setCustomTarget: (t: CustomTarget) => void
+  deleteCustomTarget: () => void
+  rank: number
+  maxRank: number
+  setTargetIndex: (ind?: number) => void
+  onDup: () => void
+}) {
+  const { t } = useTranslation('page_character')
+  const {
+    character: { key: characterKey },
+  } = useContext(CharacterContext)
+  const { data } = useContext(DataContext)
+  const {
+    path,
+    weight,
+    hitMode,
+    reaction,
+    infusionAura,
+    bonusStats,
+    description,
+  } = customTarget
+
+  const [collapse, setcollapse] = useState(true)
+
+  const setWeight = useCallback(
+    (weight: number) => setCustomTarget({ ...customTarget, weight }),
+    [customTarget, setCustomTarget]
+  )
+  const node = objPathValue(data.getDisplay(), path) as CalcResult | undefined
+  const setFilter = useCallback(
+    (bonusStats: CustomTarget['bonusStats']) =>
+      setCustomTarget({ ...customTarget, bonusStats }),
+    [customTarget, setCustomTarget]
+  )
+
+  const statEditorList = useMemo(
+    () => (
+      <StatEditorList
+        statKeys={keys}
+        statFilters={bonusStats}
+        setStatFilters={setFilter}
+        wrapperFunc={wrapperFunc}
+        label={t('addStats.label')}
+      />
+    ),
+    [bonusStats, setFilter, t]
+  )
+
+  const isMeleeAuto =
+    isCharMelee(characterKey) &&
+    (path[0] === 'normal' || path[0] === 'charged' || path[0] === 'plunging')
+  const isTransformativeReaction = path[0] === 'reaction'
+
+  const theme = useTheme()
+  const isMobile = useMediaQuery(theme.breakpoints.down('sm'))
+  return (
+    <CardThemed
+      bgt="light"
+      sx={{
+        boxShadow: '0 0 10px black',
+        position: 'sticky',
+        bottom: `10px`,
+        zIndex: 1000,
+      }}
+    >
+      <Box sx={{ display: 'flex', gap: 1, alignItems: 'center' }}>
+        <CardActionArea
+          sx={{
+            display: 'flex',
+            flexGrow: 1,
+            gap: 1,
+            height: '100%',
+            py: 1,
+            alignItems: 'center',
+          }}
+          onClick={() => setcollapse((c) => !c)}
+        >
+          <Typography variant="h6">Target Editor</Typography>
+          {collapse ? <KeyboardArrowUpIcon /> : <KeyboardArrowDownIcon />}
```

Comment:
Is there a way to expand autoamtically when you select a target? Seems like a logical move

Reply:
That makes sense.